From 4775749d81b88c85fc2ea03ad88c8034b7474b97 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 22 Sep 2023 18:31:51 +0200
Subject: [PATCH 360/954] feat(stm32mp2): add IAC defines

Add some IAC defines for STM32MP25x SoC family.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git f6343918cc5e7d569d622a35f11a721b95e0b483 feat(stm32mp2): add IAC defines]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I87375a158eab57a17bb7fe350f463c4d4b334a37
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/329839
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 24ebac7853..5e38c8950a 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -217,6 +217,20 @@ enum ddr_type {
 #define STM32MP_DEVICE_BASE		U(0x40000000)
 #define STM32MP_DEVICE_SIZE		U(0x40000000)
 
+/*******************************************************************************
+ * STM32MP2 IAC
+ ******************************************************************************/
+#define IAC_BASE			U(0x42090000)
+
+#define IAC_NB				U(6)
+#define IAC_IER0			U(0x00)
+#define IAC_ISR0			U(0x80)
+#define IAC_ICR0			U(0x100)
+
+#define IAC_IER(n)			(IAC_IER0 + (U(4) * (n)))
+#define IAC_ISR(n)			(IAC_ISR0 + (U(4) * (n)))
+#define IAC_ICR(n)			(IAC_ICR0 + (U(4) * (n)))
+
 /*******************************************************************************
  * STM32MP2 RCC
  ******************************************************************************/
-- 
2.53.0

